gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java

```diff
@@ -1599,4 +1616,307 @@ public void testAggregatedListSubnetworks() throws InterruptedException {
+    operation = remoteInstance.delete();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    assertNull(compute.get(instanceId));
+    address.delete();
+  }
+
+  @Test
+  public void testStartStopAndResetInstance() throws InterruptedException {
+    String instanceName = BASE_RESOURCE_NAME + "start-stop-reset-instance";
+    InstanceId instanceId = InstanceId.of(ZONE, instanceName);
+    NetworkId networkId = NetworkId.of("default");
+    NetworkInterface networkInterface = NetworkInterface.builder(networkId).build();
+    AttachedDisk disk = AttachedDisk.of("dev0",
+        CreateDiskConfiguration.builder(IMAGE_ID).autoDelete(true).build());
+    InstanceInfo instanceInfo =
+        InstanceInfo.builder(instanceId, MachineTypeId.of(ZONE, MACHINE_TYPE))
+            .attachedDisks(disk)
+            .networkInterfaces(networkInterface)
+            .build();
+    Operation operation = compute.create(instanceInfo);
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    Instance remoteInstance =
+        compute.get(instanceId, Compute.InstanceOption.fields(Compute.InstanceField.STATUS));
+    assertEquals(InstanceInfo.Status.RUNNING, remoteInstance.status());
+    operation = remoteInstance.stop();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    remoteInstance =
+        compute.get(instanceId, Compute.InstanceOption.fields(Compute.InstanceField.STATUS));
+    assertEquals(InstanceInfo.Status.TERMINATED, remoteInstance.status());
+    operation = remoteInstance.start();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    remoteInstance =
+        compute.get(instanceId, Compute.InstanceOption.fields(Compute.InstanceField.STATUS));
+    assertEquals(InstanceInfo.Status.RUNNING, remoteInstance.status());
+    operation = remoteInstance.reset();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    remoteInstance =
+        compute.get(instanceId, Compute.InstanceOption.fields(Compute.InstanceField.STATUS));
+    assertEquals(InstanceInfo.Status.RUNNING, remoteInstance.status());
+    remoteInstance.delete();
+  }
+
+  @Test
+  public void testSetInstanceProperties() throws InterruptedException {
+    String instanceName = BASE_RESOURCE_NAME + "set-properties-instance";
+    InstanceId instanceId = InstanceId.of(ZONE, instanceName);
+    NetworkId networkId = NetworkId.of("default");
+    NetworkInterface networkInterface = NetworkInterface.builder(networkId).build();
+    AttachedDisk disk = AttachedDisk.of("dev0",
+        CreateDiskConfiguration.builder(IMAGE_ID).autoDelete(true).build());
+    InstanceInfo instanceInfo =
+        InstanceInfo.builder(instanceId, MachineTypeId.of(ZONE, MACHINE_TYPE))
+            .attachedDisks(disk)
+            .networkInterfaces(networkInterface)
+            .build();
+    Operation operation = compute.create(instanceInfo);
+    while (!operation.isDone()) {
```

Comment:
No we don't (and also for BigQuery jobs, that have a very similar pattern). The reason is that having an `operation.waitTermination()` method that simply hangs waiting for operation to be done seemed a bit odd to me. An alternative would be:
```java
operation.whenDone(new OperationCallback() {
  public void onSuccess(Operation operation) {
    // operation succeeded, operation.errors() is null
  }

  public void onFailure(List<OperationError> errors) {
    // handle operation failure
  }
});
```

The only concern I have with this latter approach is that it gives the idea of being asynchronous while in reality it blocks until the operation is done. Any opinion?

I will open an issue for this this to we can address it also for BigQuery and do not block this PR.